### ISSUE TITLE
Enhance Effects Modifiers window

### ISSUE THREAD

**Issue by crnormand** [crnormand/gurps#1057]
I knew great minds think alike.   This is what I was talking about.   A place where relevant current information can be displayed, along with a place to take notes.    I roughed up a layout I was thinking about:

![image](https://user-images.githubusercontent.com/8931036/141195027-48d60f0d-eab3-4b23-a366-3644b5800a0f.png)

The first panel is already created.   I was just thinking of creating 2 more panels for selected tokens and/or targeted tokens to show their effects.  Or we could just expand the current panel to show:

Current Actor effects
Targeted Actors effects
Selected Actors Effects

All in the same panel (which hopefully would grown and shrink as the list changes)

Then a general editor window (like a Journal page) where a player can take notes, and embed OtFs (or drag and drop OtFs), and then a row of buttons that the user can edit, and when clicked on, will execute all of the OtFs included, like:

+2 Bonus from Buff spell
M:Broadsword

Thoughts?

**Comment by Farmeroz** [comment]
If it fits/works, one panel is better than many to reduce clutter and maintain focus, imo.  The concept above looks great.

**Comment by mjeffw** [comment]
The Effect Modifier window I've implemented is actually the selected token, not GURPS.LastActor. But there really is no difference between those two things. Active Effects exist on the actor, even if that's a synthetic actor such as an unlinked token. 

I think the prototype might be too big. Maybe for some of this we can use Token Action HUD?

**Comment by crnormand** [comment]
I agree that my original picture is way too big.   Here are the goals I am trying to achieve for a player:
1. show effects (and possible modifiers) on the player's character (usually GURPS.LastActor)
2. show effects on targeted or selected (or both) tokens.   This will let the play know of effects that can affect their roles (I am thinking of things like buff spells... specifically Grend's "distractions" buff that gives a -3 to hit.
3. Allow the play an area to enter in (or drag in) applicable OtFs, like the "distractions" buff.   Anything in this area would appear to others if they looked at your character.
4. And then finally, a place for temporary macros.   We have the list of 10 permanent macros, but maybe it would be nice to create a single button that could apply as bunch of OtFs as a single click.    The user could enter/drag in values and access them during the combat.   I could see things like a buff for damage and the damage roll for a weapon combined into a single button.   It would only be valid for that combat.

Secondly, I was thinking from the player's standpoint.   And even though I haven't run as a GM yet, I think that behaviour would work for a GM also.

**Comment by mjeffw** [comment]
I think 1 and and part of 2 are addressed by the current Effect Modifiers window. I plan to have another window or section that displays target actor modifiers that affect other players, like the distractions buff. 

The other items might be another window or section that appears on demand.

**Comment by mjeffw** [comment]
The only thing missing at this point is the editable/embeddable OTF section.
